ocaml/parsing/parser.mly

```diff
@@ -3863,16 +3864,25 @@ type_parameters:
       { ps }
 ;
 
-jkind:
-    jkind MOD mkrhs(LIDENT)+ { (* LIDENTs here are for modes *)
+jkind_parameters:
+    p = jkind_base
+      { [p] }
+  | LPAREN ps = separated_nonempty_llist(COMMA, jkind) RPAREN (* ( jkind+ )  *)
+      { ps }
+;
+
+jkind_base:
+    (* not parsing this for now as e.g. mod doesn't seem to make sense *)
+    (* LPAREN k=jkind RPAREN { k } *)
```

Comment:
This rule actually introduced parser conflicts (one shift/reduce), so that's the main reason I didn't add it. I think I had some intuition why there's a conflict when I was working on this, but I don't remember anymore.

The example you gave actually parses, I extended `parsing_arrows.ml` with it:

```
(* TEST
  expect;
*)

type p : value => value
type q : (value) => value
type r : (value => value) => value
type s : (value, value) => value
type t : (value => value, value) => value
type u : (value, value mod local) => value
type v : (value) => (value) => value

[%%expect{|
Line 1, characters 9-23:
1 | type p : value => value
             ^^^^^^^^^^^^^^
Error: Arrow jkind (=>) syntax parsed, but annotations are not implemented
|}]
```

and everything still seems to work out, unless my test isn't set up correctly (it only fails at converting the Jane_syntax annotation).

Reply:
the parsing derivation is `(value => value)` as a parameter list of length 1.

Reply:
Presumably the conflict is because you are using `separated_nonempty_list` rather than `separated_nontrivial_list`. That one would force the parenthesized list to have at least two elements, allowing the other case to handle a parenthesized single argument.

Reply:
I can't seem to get this to work without complicating the grammar (I think it can't disambiguate between the start of a list and parenthesisation). Is it fine to keep the way it is?

Reply:
Ok, leave it for now. We can sort it out later.